library/Zend/Mvc/Controller/AbstractRestfulController.php

```diff
@@ -374,8 +375,12 @@ public function onDispatch(MvcEvent $e)
                     $id = null;
                 }
                 $action = 'head';
-                $this->head($id);
+                $headResult = $this->head($id);
                 $response = $e->getResponse();
+                if ($headResult instanceof HttpResponse) {
+                    $response->setStatusCode($headResult->getStatusCode());
```

Comment:
After further inspection, this solution is a bit naive, as it doesn't copy headers and other information from the `$headResult` into the `$response`.